### ISSUE TITLE
Why my exit always loss

### ISSUE THREAD

**Issue by Vietdung113** [freqtrade/freqtrade#9266]
I have my own strategy and try to plot it. 
Here is my result. 
![image](https://github.com/freqtrade/freqtrade/assets/22492437/958bd00e-24d5-4cfd-aa4b-b255dd54079e)

The exit action is already above entry but it still gets lost although the fee is set to 0. 
How can I adjust this problem?

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

---

with the provided information - there's little to absolutely nothing we can do for you.

it's a picture - which doesn't even show if the trade shows as profit or loss, nor includes any other (even remotely) useful information, like if you're looking at live trades, or at a backtest result.

Also, i'd strongly suggest to use freqUI for this type of analysis. 
plot-dataframe  is kindof a legacy tool - which won't show you everything there is to know (a plot will never do - but freqUI will show you more).

**Comment by jackchalat** [comment]
Would be nice to see your exit logic. Based on the very little data you provided, I think you have multiple exit conditions working against one another. Maybe try using only `minimal_roi` and remove all other exit logic. See if the problem still exists.

Just a guess though. Need more info.

**Comment by Vietdung113** [comment]
> Please get used to filling out the issue template. Those who frequently omit the template may have their issues closed without comment. We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.
> 
> Filling out the template helps reduce our workload, and we greatly appreciate that.
> 
> with the provided information - there's little to absolutely nothing we can do for you.
> 
> it's a picture - which doesn't even show if the trade shows as profit or loss, nor includes any other (even remotely) useful information, like if you're looking at live trades, or at a backtest result.
> 
> Also, i'd strongly suggest to use freqUI for this type of analysis. plot-dataframe is kindof a legacy tool - which won't show you everything there is to know (a plot will never do - but freqUI will show you more).

Thank you for your recommendation. 
Here is my custom exit 
```
    def custom_exit(self, pair: str, trade: Trade, current_time: datetime, current_rate: float, current_profit: float, **kwargs) -> str | bool | None:
        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        entry_index = dataframe[dataframe["date_15m"] == trade.open_date].index
        if len(entry_index) == 0:
            return None
        else:
            entry_index = entry_index[0]

        last_5_candle_before_entry = dataframe.iloc[entry_index - 5: entry_index]
        stop_loss = last_5_candle_before_entry["low_15m"].min()
        if trade.open_rate < stop_loss:
            stop_loss = trade.open_rate - 50
        stop_profit = trade.open_rate + (trade.open_rate - stop_loss)

        ema8_15m = dataframe["ema8_15m"].iloc[-1]
        ema13_15m = dataframe["ema13_15m"].iloc[-1]
        ema21_15m = dataframe["ema21_15m"].iloc[-1]
        if current_rate < stop_loss or current_rate > stop_profit or ema8_15m < ema13_15m or ema13_15m < ema21_15m:
            return "sell_signal"
        return None
 ```

**Comment by Vietdung113** [comment]
> Would be nice to see your exit logic. Based on the very little data you provided, I think you have multiple exit conditions working against one another. Maybe try using only `minimal_roi` and remove all other exit logic. See if the problem still exists.
> 
> Just a guess though. Need more info.

I mentioned my exit logic in the previous comment. 
Could you give me any solutions?

**Comment by jackchalat** [comment]
I can't tell what's wrong. But I'd split this part :

```
if current_rate < stop_loss or current_rate > stop_profit or ema8_15m < ema13_15m or ema13_15m < ema21_15m:
    return "sell_signal"
```
 to several exits with descriptive labels like this:

```
if current_rate < stop_loss:
   return "sell_stop_loss"
if current_rate  > stop_profit:
   return "sell_stop_profit"
```

This will help track down which condition actually exits with loss. 

I think it may have to do with your timeframe. Since your indicators have `_15m` suffix, I assume your timeframe is lower. If you are using (for example) 5m timeframe with 15m informative data, it will take 3 5m dataframes to count as 1 15m dataframe. So if you are trying to reference one previous 15m dataframe, you actually have to go back by 3. This might not be the problem but it's worth looking into.

**Comment by Vietdung113** [comment]
I've debugged and found that the fee I set on config was 0.02 (2%) and it was calculated following my exit rule with profit always < 2% so exit always loss. 
I'll close the threads. 
Thank you for all your support.
